src/coreclr/jit/codegenlinear.cpp

```diff
@@ -738,13 +730,28 @@ void CodeGen::genCodeForBBlist()
 
 #endif // !FEATURE_EH_FUNCLETS
 
-            case BBJ_NONE:
             case BBJ_SWITCH:
                 break;
 
             case BBJ_ALWAYS:
-#ifdef TARGET_XARCH
             {
+                // Peephole optimization: If this block jumps to the next one, skip emitting the jump
+                // (unless we are jumping between hot/cold sections, or if we need the jump for EH reasons)
+                // (Skip this if optimizations are disabled, unless the block shouldn't have a jump in the first place)
+                const bool tryJumpOpt =
+                    compiler->opts.OptimizationEnabled() || ((block->bbFlags & BBF_NONE_QUIRK) != 0);
+                const bool skipJump = tryJumpOpt && block->JumpsToNext() &&
+                                      ((block->bbFlags & BBF_KEEP_BBJ_ALWAYS) == 0) &&
+                                      !compiler->fgInDifferentRegions(block, block->GetJumpDest());
+                if (skipJump)
```

Comment:
`emitNop` is likely almost always false, so maybe rearrange the logic here and check for that first, and then only if true, compute `tryJumpOpt` and `skipJump`?

Might save a tiny bit of TP

Reply:
Ah, I forgot that the `emitNop` logic is specific to AMD64, so rearranging that logic caused massive size regressions. I'm still interested in rearranging the peephole opt a bit to at least save some TP on AMD64 -- this will introduce some conditional compilation, which isn't ideal, but I think it will be worth not having to check `emitNop` on any other platform.

Also, I think it's possible in many cases for `emitNop` to be false and `skipJump` to be true, right? Rearranging the two might incur big size diffs.

Reply:
Ah, right. This was a bad suggestion on my part, I didn't pay enough attention to what that `break;` was doing in there.

You should go back to the way you had things.